src/common/components/proposal-votes/index.tsx

```diff
@@ -83,67 +142,151 @@ export class ProposalVotesDetail extends BaseComponent<Props, State> {
             };
           })
           .sort((a, b) => (b.totalHp > a.totalHp ? 1 : -1));
-
-        this.stateSet({ voters, loading: false });
+        let newOriginalVoter: Voter[] = this.state.originalVoters.concat(voters);
+        const uniqueOriginalVoters = [
+          ...new Map(newOriginalVoter.map((item) => [item["name"], item])).values()
+        ];
+        if (voter) {
+          let preVoters: Voter[] = this.state.voters;
+          voters = preVoters.concat(voters);
+        }
+        this.stateSet(
+          {
+            voters,
+            originalVoters: uniqueOriginalVoters,
+            loading: false
+          },
+          this.search
+        );
       });
   };
 
+  handlePageChange = () => {
+    const { noOfPages, page, lastDataLength, limit } = this.state;
+    if (page === noOfPages && lastDataLength === limit) {
+      this.load();
+    }
+  };
   render() {
-    const { loading, voters } = this.state;
-
-    if (loading) {
+    const { loading, voters, page, sort, originalVoters } = this.state;
+    if (loading && !voters.length && !originalVoters.length) {
       return <LinearProgress />;
     }
+    const pageSize = 12;
+    const start = (page - 1) * pageSize;
+    const end = start + pageSize;
+    const sliced = voters
+      .sort((a, b) => {
+        const keyA = a[sort]!;
+        const keyB = b[sort]!;
+
+        if (keyA > keyB) return -1;
+        if (keyA < keyB) return 1;
+        return 0;
```

Comment:
I guess you can simplify it: `voters.sort((a,b) => a[sort] - b[sort])`

Reply:
If we use this method. There is a TS error as it could not recognize what would be the value of a[sort], because a is an object and could not determine what key would be there in sort. Therefore we could not use this method. 

However I have tried summing this up with simple logical operator such as voters.sort((a,b) => +(a[sort] > b[sort])). But we would miss the case where a[sort] == b[sort]. Therefore I think we should keep this as it is. Please suggest if you have any other solution in mind.